firefox_puppeteer/api/prefs.py

```diff
@@ -202,3 +202,13 @@ def set_pref(self, pref_name, value):
             """, script_args=[pref_name, value])
 
         assert retval
+
+    def persist_prefs(self):
+        """Persist current pref values to the default prefs file (typically prefs.js)
+        """
+        with self.marionette.using_context('chrome'):
+            self.marionette.execute_script("""
+              Cu = Components.utils;
+              Cu.import("resource://gre/modules/Services.jsm");
+              Services.prefs.savePrefFile(null);
```

Comment:
Hm, it doesn't look like that this is really the problem here. `tearDown` is really only called at the very end of the test. So I don't know why Marionette resets those preferences too.